### PR TITLE
feat(route): add HAFU 河南财政金融学院

### PR DESCRIPTION
**Pull request by deep1nlife**
## 该 PR 相关 Issue / Involved issue

Close #10245 

## 完整路由地址 / Example for the proposed route(s)

```routes
/hafu/news/ggtz
/hafu/news/jwc
/hafu/news/zsjyc
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [X] 新的路由 New Route
  - [X] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [X] 文档说明 Documentation
  - [X] 中文文档 CN
  - [ ] 英文文档 EN
- [X] 全文获取 fulltext
  - [X] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [X] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [X] 可以解析 Parsed
  - [X] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
@TonyRL Thanks for your patience in reviewing my previous crappy code on PR #10245 , now that I've rewritten them...

### CODE REVIEW

**Comment by WenPeiTung** [comment]
> Successfully [generated](https://github.com/DIYgod/RSSHub/actions/runs/2740798384) as following:
> http://localhost:1200/hafu/news/:type/:limit? - Failed

The optional key word 'limit' that value is empty in the 'Route requested', that's normal? Do I need to fix it?

**Comment by WenPeiTung** [comment]
> Successfully [generated](https://github.com/DIYgod/RSSHub/actions/runs/2740798384) as following:
> http://localhost:1200/hafu/news/:type/:limit? - Failed

I was confused last night, I updated the route example, please test it again.

File: lib/v2/hafu/router.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = function (router) {
+    router.get('/news/:type/:limit?', require('./news'));
```
Comment (TonyRL) [inline lib/v2/hafu/router.js:2]:
```suggestion
    router.get('/news/:type?', require('./news'));
```

File: lib/v2/hafu/utils.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+function ggtzParse(ctx, $) {
+    const data = $('a[class=c269582]').parent().slice(0, Limit);
+    const resultItems = data
+        .map(async (_, item) => {
+            const title = $(item).find('a[class=c269582]').attr('title');
+            // .slice(3) for cut out str '../' in original link
+            const href = $(item).find('a[class=c269582]').attr('href').slice(3);
+            const link = typeMap.ggtz[1] + href;
+
+            const cache = await ctx.cache.get(link);
```
Comment (TonyRL) [inline lib/v2/hafu/utils.js:45]: Please use ctx.cache.tryGet()
https://docs.rsshub.app/joinus/quick-start.html#ti-jiao-xin-de-rsshub-gui-ze-bian-xie-jiao-ben-shi-yong-huan-cun

File: lib/v2/hafu/utils.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+            const result = {
+                title,
+                description,
+                pubDate: timezone(pubDate, +8),
+                link,
+                author,
+            };
+            //                                                 cache expiretime 120mins
+            ctx.cache.set(link, JSON.stringify(result), config.cache.contentExpire * 24);
```
Comment (TonyRL) [inline lib/v2/hafu/utils.js:77]: Default value is enough

File: lib/v2/hafu/utils.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                description = art(path.join(__dirname, 'templates/hafu.art'), articleBody)();
+            } catch {
+                description = href;
+            }
+            const header = articleData('h1').next().text();
+            const index = header.indexOf('日期');
+
+            const author = header.substring(0, index - 2) || '';
+            // form articlePage || hostPage
+            const Date = header.substring(index + 3, index + 19) || $(item).find('a[class=c269582_date]');
```
Comment (TonyRL) [inline lib/v2/hafu/utils.js:66]:
```suggestion
            const date = header.substring(index + 3, index + 19) || $(item).find('a[class=c269582_date]').text();
```
camelCase
Missing `.text()`

File: lib/v2/hafu/utils.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+            const pubDate = parseDate(Date, 'YYYY-MM-DD HH-mm') || parseDate(Date, 'YYYY-MM-DD  ');
```
Comment (TonyRL) [inline lib/v2/hafu/utils.js:67]:
```suggestion
            const pubDate = parseDate(Date, 'YYYY-MM-DD HH:mm') || parseDate(Date, 'YYYY-MM-DD');
```

File: lib/v2/hafu/templates/hafu.art [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{{ if articleBody }}
+    {{@ articleBody }}
+{{ /if }}
```
Comment (TonyRL) [inline lib/v2/hafu/templates/hafu.art:3]: I think user will be happier if you can add attachments

File: docs/university.md [hunk 7]
```diff
@@ -1252,6 +1252,18 @@ category 列表：
 
 <Route author="plusmultiply0" example="/hhu/libNewsc" path="/hhu/libNewsc"/>
 
+## 河南财政金融学院
+
+### 河南财政金融学院
+
+\<Route author="WenPeiTung" example="/hafu/news/ggtz" path="/hafu/news/:type?" radar="1" rssbud="1" :paramsDesc="\[' 分类，见下表（默认为"ggtz")',' 支持全局参数"limit"，默认为"10"']">
```
Comment (TonyRL) [inline docs/university.md:1259]:
```suggestion
<Route author="WenPeiTung" example="/hafu/news/ggtz" path="/hafu/news/:type?" radar="1" rssbud="1" :paramsDesc="['分类，见下表（默认为 `ggtz`)']">
```

File: lib/v2/hafu/utils.js [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+            });
+    }
+
+    return articleBody;
+}
+// A. got from hostPage     1.article(link), 2.article(title), 3.(pubDate)
+// B. got from articlePage  1.description(fullText), 2.article(author), 3.detailed(pubDate)
+function ggtzParse(ctx, $) {
+    const data = $('a[class=c269582]').parent().slice(0, limit);
+    const resultItems = data
```
Comment (TonyRL) [inline lib/v2/hafu/utils.js:87]:
```suggestion
async function ggtzParse(ctx, $) {
    const data = $('a[class=c269582]').parent().slice(0, limit);
    const resultItems = await Promise.all(data.map(...));
    return resultItems;
}
```

File: lib/v2/hafu/utils.js [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+
+module.exports = async (ctx, type) => {
+    const link = typeMap[type].url;
+    const title = typeMap[type].title;
+
+    const response = await got(link);
+    const $ = cheerio.load(response.data);
+
+    limit = ctx.query.limit || limit;
+    const resultList = typeMap[type].parseFn(ctx, $);
```
Comment (TonyRL) [inline lib/v2/hafu/utils.js:24]:
```suggestion
    const resultList = await typeMap[type].parseFn(ctx, $);
```

**Comment by WenPeiTung** [comment]
finally...

**Comment by WenPeiTung** [comment]
finally...
I added so much trouble for you, I feel sorry, I'll express my sincerity again, thank you！

**Pull request merged**